Support for `blobIndexMatch` in `azurerm_storage_management_policy` (#10400)
terraform-providers/terraform-provider-azurerm: website/docs/r/storage_management_policy.html.markdown

* `prefix_match` - An array of strings for prefixes to be matched.
* `blob_types` - An array of predefined values. Valid options are `blockBlob` and `appendBlob`.

> * `blob_index_match` - A `blob_index_match` block as defined below.

Reviewer: could we expand what "blob index matching" does?

> ~> **NOTE:** Please enable `blobIndex` feature by [PSH or Cli commands](https://azure.microsoft.com/en-us/blog/manage-and-find-data-with-blob-index-for-azure-storage-now-in-preview/) before setting the block `blob_index_match`. 

Reviewer: grammer
```suggestion
~> **NOTE:** This property requires enabling the `blobIndex` feature with [PSH or Cli commands](https://azure.microsoft.com/en-us/blog/manage-and-find-data-with-blob-index-for-azure-storage-now-in-preview/). 
```

* `delete_after_days_since_creation_greater_than` - The age in days after create to delete the snaphot. Must be at least 0.

---

`blob_index_match` supports the following:

* `tag_name` - The age in days after create to delete the snaphot. Must be at least 0.
* `tag_op` - The age in days after create to delete the snaphot. Must be at least 0.

> * `tag_value` - The age in days after create to delete the snaphot. Must be at least 0.

Reviewer: These descriptions are wrong?
